crates/proto/README.md

```diff
@@ -1,9 +1,68 @@
+`hickory-proto` is a safe and secure low-level DNS library. 
+This is the foundational DNS protocol library used the other higher-level HickoryDNS crates.
 
-**NOTICE** This project was rebranded from Trust-DNS to Hickory DNS and has been moved to the https://github.com/hickory-dns/hickory-dns organization and repo, this crate/binary has been moved to [hickory-proto](https://crates.io/crates/hickory-proto), from `0.24` and onward, for prior versions see [trust-dns-proto](https://crates.io/crates/trust-dns-proto).
+Unless you want to manipulate the DNS packets directly, it is likely not the library you want.
+For higher-level interfaces, refer to the [`hickory-server`], [`hickory-client`],
+[`hickory-resolver`] and [`hickory-recursor`] library crates instead.
 
-## Versioning
+[`hickory-server`]: ../server
+[`hickory-client`]: ../client
+[`hickory-resolver`]: ../resolver
+[`hickory-recursor`]: ../recursor
 
-Hickory DNS does it's best job to follow semver. Hickory DNS will be promoted to 1.0 upon stabilization of the publicly exposed APIs. This does not mean that Hickory DNS will necessarily break on upgrades between 0.x updates. Whenever possible, old APIs will be deprecated with notes on what replaced those deprecations. Hickory DNS will make a best effort to never break software which depends on it due to API changes, though this can not be guaranteed. Deprecated interfaces will be maintained for at minimum one major release after that in which they were deprecated (where possible), with the exception of the upgrade to 1.0 where all deprecated interfaces will be planned to be removed.
+## Cryptography provider
+
+The `hickory-proto` library supports choosing between two cryptography providers for
+protocols that require cryptography (DNSSEC, DNS-over-TLS, etc):
+
+1. [aws-lc-rs]
+2. [ring]
+
+Other feature flags are suffixed by the choice of provider. For example `tls-ring` or
+`tls-aws-lc-rs`.
+
+[aws-lc-rs]: https://crates.io/crates/aws-lc-rs
+[ring]: https://crates.io/crates/
+
+## DNSSEC
+
+In order to use DNSSEC you must enable a cargo feature corresponding to your
+desired cryptography provider ([aws-lc-rs] or [ring]):
+
+* `dnssec-aws-lc-rs`
+* `dnssec-ring`
+
+## Protocol support
+
+The following DNS protocols are optionally supported:
+
+* DNS over TLS (DoT)
+* DNS over HTTP/2 (DoH)
+* DNS over QUIC (DoQ)
+* DNS over HTTP/3 (DoH3)
+
+In order to use these optional protocols you must enable a cargo feature
+corresponding to your desired cryptography provider ([aws-lc-rs] or [ring]):
+
+* DoT: `tls-aws-lc-rs` or `tls-ring`.
+* DoH: `https-aws-lc-rs` or `https-ring`
+* DoQ: `quic-aws-lc-rs` or `quic-ring`
+* DoH3: `h3-aws-lc-rs` or `h3-ring`
+
+## Other crate features
+
+* `text-parsing` - support for parsing trust anchor DNSKEY records from a file.
```

Comment:
This is also for parsing zone files more generally, since it also gates `hickory_proto::serialize::txt`, which includes `RDataParser`, etc.
```suggestion
* `text-parsing` - support for reading text-based zone files.
```